google-beta/resource_bigquery_table.go

```diff
@@ -62,6 +63,182 @@ func resourceBigQueryTable() *schema.Resource {
 				Computed: true,
 			},
 
+			// ExternalDataConfiguration [Optional] Describes the data format,
+			// location, and other properties of a table stored outside of BigQuery.
+			// By defining these properties, the data source can then be queried as
+			// if it were a standard BigQuery table.
+			"external_data_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// Autodetect : [Required] If true, let BigQuery try to autodetect the
+						// schema and format of the table.
+						"autodetect": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						// BigtableOptions: [Optional] Additional options if sourceFormat is set to BIGTABLE.
+						"bigtable_options": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									// ColumnFamilies: [Optional] List of column families to expose in the
+									// table schema along with their types.
+									"column_families": {
```

Comment:
I think this can be expressed in Terraform schema; is there a particular reason you preferred JSON here?

Reply:
This would have been quite a big piece to implement so I just went with JSON here since the `schema` for tables already does that.

I think this and `external_data.configuration.schema` and `bigquery_table.schema` all should be on the same format, maybe they should all be changed, but that should probably be a separate change and I'm not sure how to handle such breaking changes? What do you think?

Reply:
I think the `schema` objects are a good place to draw the line but this object seems appropriate to implement in Terraform.

Also, if the field is optional and it isn't something you need configured, feel free to omit it entirely for now- we can always add it later based on a user's demand.

Reply:
I'll probably remove BigTable support from this PR and we can add it in a separate PR. (It's also a bit more expensive to test, and I have to ensure I don't forget to teardown the BT cluster between testing sessions 💸)

Reply:
Sounds good to me!

Reply:
Dropped _Bigtable_ support in https://github.com/GoogleCloudPlatform/magic-modules/pull/1466/commits/401502b6dfe90251cae30fd7f266f034807ef88e.